docs/source/10_deployment/12_dask.md

```diff
@@ -0,0 +1,313 @@
+
+```python
+from .dask_runner import DaskRunner
+
+__all__ = ["DaskRunner"]
+```
+
+Copy the contents of the script below into `dask_runner.py`:
+
+```python
+"""``DaskRunner`` is an ``AbstractRunner`` implementation. It can be
+used to distribute execution of ``Node``s in the ``Pipeline`` across
+a Dask cluster, taking into account the inter-``Node`` dependencies.
+"""
+from collections import Counter
+from itertools import chain
+from typing import Any, Dict
+
+from distributed import Client, as_completed, worker_client
+from kedro.io import AbstractDataSet, DataCatalog
+from kedro.pipeline import Pipeline
+from kedro.pipeline.node import Node
+from kedro.runner import AbstractRunner, run_node
+
+
+class _DaskDataSet(AbstractDataSet):
+    """``_DaskDataSet`` publishes/gets named datasets to/from the Dask
+    scheduler."""
+
+    def __init__(self, name: str):
+        self._name = name
+
+    def _load(self) -> Any:
+        try:
+            with worker_client() as client:
+                return client.get_dataset(self._name)
+        except ValueError:
+            # Upon successfully executing the pipeline, the runner loads
+            # free outputs on the scheduler (as opposed to on a worker).
+            Client.current().get_dataset(self._name)
+
+    def _save(self, data: Any) -> None:
+        with worker_client() as client:
+            client.publish_dataset(data, name=self._name, override=True)
+
+    def _exists(self) -> bool:
+        return self._name in Client.current().list_datasets()
+
+    def _release(self) -> None:
+        Client.current().unpublish_dataset(self._name)
+
+    def _describe(self) -> Dict[str, Any]:
+        return dict(name=self._name)
+
+
+class DaskRunner(AbstractRunner):
+    """``DaskRunner`` is an ``AbstractRunner`` implementation. It can be
+    used to distribute execution of ``Node``s in the ``Pipeline`` across
+    a Dask cluster, taking into account the inter-``Node`` dependencies.
+    """
+
+    def __init__(self, client_args: Dict[str, Any] = {}, is_async: bool = False):
+        """Instantiates the runner by creating a ``distributed.Client``.
+
+        Args:
+            client_args: Arguments to pass to the ``distributed.Client``
+                constructor.
+            is_async: If True, the node inputs and outputs are loaded and saved
+                asynchronously with threads. Defaults to False.
+        """
+        super().__init__(is_async=is_async)
+        Client(**client_args)
+
+    def __del__(self):
+        Client.current().close()
+
+    def create_default_data_set(self, ds_name: str) -> _DaskDataSet:
+        """Factory method for creating the default dataset for the runner.
+
+        Args:
+            ds_name: Name of the missing dataset.
+
+        Returns:
+            An instance of ``_DaskDataSet`` to be used for all
+            unregistered datasets.
+        """
+        return _DaskDataSet(ds_name)
+
+    def _run_node(
```

Comment:
Does this need to be a class method, or can it be a standalone function? As far as I can tell it's just a thin wrapper around `run_node` so that you pass `dependencies`, but they're not actually used here. Presumable the dask client is?

Reply:
@lorenabalan It doesn't need to be an instance method. I could make it into a static method, if so desired?  It could also be outside the class, but I think:

1. grouping for functionality is valid reason to make it into a static method
2. for the purpose of a deployment guide, having something outside the class is harder to keep track of (not really, but arguably)

Open to doing whatever with this; let me know if you have a strong preference. I can leave it as is or make it into a static method, with least preference for moving it outside the class, but can roll with anything.

Reply:
Yeah fair enough about the grouping aspect. Let's stick a `staticmethod` for this one.

Reply:
Done!